src/PKSim.Infrastructure/ORM/Queries/ModelObserverQuery.cs

```diff
@@ -67,6 +68,13 @@ public IObserverBuildingBlock AllObserversFor(IMoleculeBuildingBlock moleculeBui
          return observerBuildingBlock;
       }
 
+      private void addSimulationObservers(Simulation simulation, IObserverBuildingBlock observerBuildingBlock)
+      {
+         simulation.AllBuildingBlocks<ObserverSet>()
```

Comment:
Don't care at that stage about duplicated name etc.. DEpending on condition (For all moelcule etc..) it could still va valid

Reply:
Also I don't think I should clone the observer here. They are already specific to the simulation AND They won't be persisted anywhere  as core building blocks are discarded

Reply:
So in case we have 2 observers with the same name (and Container-Criteria will allow creation of both in the same parent container), build process will crash with "not unique" exception?

Reply:
yes. Name "MyObserer" is not unique in parent container "XXx"

I would love if the error would use a type instead of parent container but this is a good as it gets for now